compiler/rustc_mir_dataflow/src/impls/initialized.rs

```diff
@@ -50,15 +50,19 @@ use crate::{lattice, AnalysisDomain, GenKill, GenKillAnalysis, MaybeReachable};
 /// Similarly, at a given `drop` statement, the set-intersection
 /// between this data and `MaybeUninitializedPlaces` yields the set of
 /// places that would require a dynamic drop-flag at that statement.
-pub struct MaybeInitializedPlaces<'a, 'tcx> {
+pub struct MaybeInitializedPlaces<'a, 'mir, 'tcx> {
     tcx: TyCtxt<'tcx>,
-    body: &'a Body<'tcx>,
+    body: &'mir Body<'tcx>,
     mdpe: &'a MoveDataParamEnv<'tcx>,
     skip_unreachable_unwind: bool,
 }
 
-impl<'a, 'tcx> MaybeInitializedPlaces<'a, 'tcx> {
-    pub fn new(tcx: TyCtxt<'tcx>, body: &'a Body<'tcx>, mdpe: &'a MoveDataParamEnv<'tcx>) -> Self {
+impl<'a, 'mir, 'tcx> MaybeInitializedPlaces<'a, 'mir, 'tcx> {
```

Comment:
I will make a plea for better lifetime names than `'a`. In a type with multiple lifetimes, I don't like when some of the names are useful (`'mir`) and some are not (`'a`).  `Elaborator<'a, 'b, 'mir, 'tcx>` is a good example.

I know some of the `'a` names are pre-existing but we can do better for the new ones. Would `'mdpe` be a good name here instead of `'a`? Or `'env`? Anything even slightly descriptive would be better than `'a`. And I think that would propagate to a lot of the other changes in this commit.